### PR TITLE
Use real DOM for testing (2 of 2).

### LINKED ISSUE #178: Use real DOM for testing
The resource and response tests rely on a FakeElement object that has its own implementation of DOM methods like appendChild and insertBefore. This is fine for the common case, but edge cases like older version of Internet Explorer may have quirks that are not reflected in the FakeElement implementation. This masks bugs that would otherwise fail the tests.

These tests should instead rely on the native DOM and its methods.

### PR DESCRIPTION
**Pull request by rviscomi**
Resources are created in an iframe using real DOM elements as opposed to FakeElements.
el.href would return the canonical URL ("file://foo.bar") so we use el.getAttribute('href') instead ("//foo.bar").

Closes #178

### CODE REVIEW

File: src/client/net/resource_test.js [hunk 1]
```diff
@@ -81,70 +56,37 @@ describe('spf.net.resource', function() {
...
+      return a.getAttribute('href');
+    });
   };
 
 
   beforeEach(function() {
+    // Test resources in an iframe. Use that as the document for all tests.
+    iframe = document.createElement('iframe');
+    document.body.appendChild(iframe);
+    fakes.doc = iframe.contentWindow.document;
```
Comment (rviscomi) [inline src/client/net/resource_test.js]: IE stumbles here. Working on another approach.

**Comment by rviscomi** [comment]
I've been pulled away from this and will need more time to resolve the IE problem. Closing this PR for now and will resubmit once a fix is ready.
